### ISSUE TITLE
Bug: id not allowed in RUN mount from buildkit

### ISSUE THREAD

**Issue by anoldguy** [hadolint/hadolint#433]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

(Could be related to #432, as it's the new parser)

### Expected behavior

Hadolint allows me to tag my cache mount with an appropriate id, as allowed by [the buildkit docs](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/experimental.md#run---mounttypecache).

```
RUN --mount=id=apt-cache,type=cache,target=/var/cache/apt apt-get update
```

### Actual behavior

Hadolint tosses an error about an unexpected arg, `id=<name>`.

```
Dockerfile:16:13 unexpected "id=apt-cache" expecting "from=", "readonly", 
  "readwrite", "ro", "rw", "type=", source=, or target=
```

### Steps to reproduce the behavior

When specifying the cache for the RUN command, name it with `id=<foo>`:

```Dockerfile
RUN --mount=id=apt-cache,type=cache,target=/var/cache/apt apt-get update
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

1.17.7

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:experimental
FROM ubuntu:20.04 as base
# Subsequent RUN use pipes, so ensure we pipefail
# https://github.com/hadolint/hadolint/wiki/DL4006
SHELL [ "/bin/bash", "-o", "pipefail", "-c" ]
RUN --mount=id=apt-cache,type=cache,target=/var/cache/apt apt-get update
```

**Additional environment details (OS, stack version, etc.)**
Running under our CI, base OS Linux, but dockerized.

**Comment by lorenzo** [comment]
Heh, never expected to this this ticket this fast. This is the short story:

Parsing those flags correctly is quite complex. It is so complex that the docker parser doesn’t not report errors correctly. I made a choice in order to show better parsing errors: I would only allow mount flags if the type is the first modifier listed. Anything else is an error. 

I scouted hundreds of dockerfiles in GitHub and could not find one where the type was not declared as the first modifier. You are the first case I know of, congrats!

To solve the issue, without degrading the quality of parsing errors, I suggest that you adapt to how almost everyone else are writing that line: 
 
`mount=type=cache,id=foo`

**Comment by lorenzo** [comment]
I will also improve the error message in the parser to show that this decision was made.

**Comment by anoldguy** [comment]
Oh! That’s a fantastic response. 👍 to the workaround, too. Easy enough on our end.
